optimizers/llm_optimizer.py

```diff
@@ -9,28 +9,21 @@ def __init__(self, benchmarker, openai_api_key):
         self.client = OpenAI(api_key=openai_api_key)
 
     def optimize(self, iterations, **kwargs):
-        for _ in range(iterations):
+        for iteration in range(iterations):
             prompt = self.generate_prompt(self.history)
-            response = self.client.chat.completions.create(
-                model="gpt-3.5-turbo",
-                messages=[{"role": "system", "content": "You are a machine learning expert."}, {"role": "user", "content": prompt}],
-                temperature=0.7, max_tokens=150, top_p=1
-            )
-
-            ifDecodeError = False
-            while not ifDecodeError:
+            config = None
+            while config is None:
+                response = self.client.chat.completions.create(
+                    model="gpt-3.5-turbo",
+                    messages=[{"role": "system", "content": "You are a machine learning expert."},
+                              {"role": "user", "content": prompt}],
+                    temperature=0.7, max_tokens=150, top_p=1)
                 try:
                     config = json.loads(response.choices[0].message.content.strip())
-                    ifDecodeError = True
                 except json.decoder.JSONDecodeError:
-                    response = self.client.chat.completions.create(
```

Comment:
we need a try and except, otherwise it will throw exception and stop our experiment. Please change it back

Reply:
@roshansridhar

Reply:
thanks for adding that, its still there in line 21 and 23